kernel/src/utilities/capability_ptr.rs

```diff
@@ -0,0 +1,131 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
```

Comment:
I think this module is good except it needs more robust documentation.

@LawrenceEsswood I will take a stab at this if you don't do it before me.

Reply:
Adding comments to all methods on CapabilityPtr, and the permissions Enum.